go/types/struct.go

```diff
@@ -331,32 +333,18 @@ func (s Struct) IsZeroValue() bool {
 // Delete returns a new struct where the field name has been removed.
 // If name is not an existing field in the struct then the current struct is returned.
 func (s Struct) Delete(n string) Struct {
-	dec := s.decoder()
-	w := newBinaryNomsWriter()
-	StructKind.writeTo(&w)
-	dec.skipKind()
-	dec.copyString(&w)
-	count := dec.readCount()
-	w.writeCount(count - 1) // If not found we just return s
-
-	found := false
-	for i := uint64(0); i < count; i++ {
-		name := dec.readString()
-
-		if n == name {
-			dec.skipValue()
-			found = true
-		} else {
-			w.writeString(name)
-			dec.copyValue(&w)
-		}
-	}
+	prolog, head, tail, count, found := s.splitFieldsAt(n)
 
+	w := binaryNomsWriter{make([]byte, len(s.buff)), 0}
+	w.writeRaw(prolog)
 	if found {
```

Comment:
shouldn't this just `return s if !found`?

Reply:
yup